test/common/common/log_verbosity_update_test.cc

```diff
@@ -0,0 +1,229 @@
+  file_path.remove_suffix(3);
+  std::pair<absl::string_view, int> update = std::make_pair(file_path, 1);
+  getFancyContext().updateVerbositySetting({update});
+
+  FANCY_LOG(debug, "Debug: now level is debug");
+  SpdLoggerSharedPtr p = getFancyContext().getFancyLogEntry(__FILE__);
+  EXPECT_NE(p, nullptr);
+  EXPECT_EQ(p->level(), spdlog::level::debug);
+}
+
+TEST(Fancy, updateBasename) {
+  Logger::Context::enableFancyLogger();
+  getFancyContext().setFancyLogger(__FILE__, spdlog::level::info);
+  FANCY_LOG(info, "Info: verbosityUpdateCurrentFile test begins.");
+  FANCY_LOG(debug, "Debug: you shouldn't see this message!");
+
+  absl::string_view file_path = __FILE__;
+  file_path.remove_suffix(3);
+  const size_t position = file_path.rfind('/');
+  file_path.remove_prefix(position + 1);
+
+  std::pair<absl::string_view, int> update = std::make_pair(file_path, 1);
+  getFancyContext().updateVerbositySetting({update});
+
+  FANCY_LOG(debug, "Debug: now level is debug");
+  SpdLoggerSharedPtr p = getFancyContext().getFancyLogEntry(__FILE__);
+  EXPECT_NE(p, nullptr);
+  EXPECT_EQ(p->level(), spdlog::level::debug);
+}
+
+TEST(Fancy, multipleUpdatesBasename) {
+  const char* file_1 = "envoy/src/foo.cc";
+  const char* file_2 = "envoy/src/bar.cc";
+  std::atomic<spdlog::logger*> flogger{nullptr};
+  getFancyContext().initFancyLogger(file_1, flogger);
+  getFancyContext().initFancyLogger(file_2, flogger);
+
+  std::pair<absl::string_view, int> update_1 = std::make_pair("foo", 1);
+  std::pair<absl::string_view, int> update_2 = std::make_pair("bar", 3);
+  getFancyContext().updateVerbositySetting({update_1, update_2});
+
+  SpdLoggerSharedPtr p = getFancyContext().getFancyLogEntry(file_1);
+  EXPECT_NE(p, nullptr);
+  EXPECT_EQ(p->level(), spdlog::level::debug);
+  p = getFancyContext().getFancyLogEntry(file_2);
+  EXPECT_NE(p, nullptr);
+  EXPECT_EQ(p->level(), spdlog::level::warn);
+}
+
+TEST(Fancy, multipleMatchesBasename) {
+  const char* file_1 = "envoy/src/a.cc";
+  const char* file_2 = "envoy/a.cc";
+  std::atomic<spdlog::logger*> flogger{nullptr};
+  getFancyContext().initFancyLogger(file_1, flogger);
+  getFancyContext().initFancyLogger(file_2, flogger);
+
+  std::pair<absl::string_view, int> update_1 = std::make_pair("a", 3);
+  getFancyContext().updateVerbositySetting({update_1});
+
+  SpdLoggerSharedPtr p = getFancyContext().getFancyLogEntry(file_1);
+  EXPECT_NE(p, nullptr);
+  EXPECT_EQ(p->level(), spdlog::level::warn);
+  p = getFancyContext().getFancyLogEntry(file_2);
+  EXPECT_NE(p, nullptr);
+  EXPECT_EQ(p->level(), spdlog::level::warn);
+}
+
+TEST(Fancy, globStarUpdate) {
+  const char* file_1 = "envoy/src/foo.cc";
+  const char* file_2 = "envoy/src/bar.cc";
+  const char* file_3 = "envoy/baz.cc";
+  std::atomic<spdlog::logger*> flogger{nullptr};
+  getFancyContext().initFancyLogger(file_1, flogger);
+  getFancyContext().initFancyLogger(file_2, flogger);
+  getFancyContext().initFancyLogger(file_3, flogger);
```

Comment:
likewise as below

Reply:
ah, it does clobber the flogger value, but we end up logger pointer in the map for the others. So it's fine.